python/ray/air/tests/test_air_usage.py

```diff
@@ -0,0 +1,115 @@
+"""Unit tests for AIR telemetry."""
+
+import json
+
+import pytest
+from unittest.mock import MagicMock
+
+import ray
+from ray._private.usage.usage_lib import TagKey
+from ray.air._internal import usage as air_usage
+from ray.air.integrations import wandb, mlflow, comet
+from ray.tune.callback import Callback
+from ray.tune.logger import LoggerCallback
+from ray.tune.logger.aim import AimLoggerCallback
+from ray.tune.utils.callback import DEFAULT_CALLBACK_CLASSES
+
+
+def _mock_record_from_module(module, monkeypatch):
+    recorded = {}
+
+    def mock_record_extra_usage_tag(key: TagKey, value: str):
+        recorded[key] = value
+
+    monkeypatch.setattr(
+        module,
+        "record_extra_usage_tag",
+        mock_record_extra_usage_tag,
+    )
+    return recorded
+
+
+@pytest.fixture
+def mock_record(monkeypatch):
+    import ray.air._internal.usage
+
+    yield _mock_record_from_module(ray.air._internal.usage, monkeypatch=monkeypatch)
+
+
+@pytest.fixture(scope="module")
+def ray_start_2_cpus():
+    address_info = ray.init(num_cpus=2)
+    yield address_info
+    ray.shutdown()
+
+
+class _CustomLoggerCallback(LoggerCallback):
+    pass
+
+
+class _CustomCallback(Callback):
+    pass
+
+
+def test_tag_setup_wandb(monkeypatch):
+    from ray.air.integrations.wandb import _setup_wandb
+
+    recorded = _mock_record_from_module(
+        ray.air.integrations.wandb, monkeypatch=monkeypatch
+    )
+    _setup_wandb(trial_id="a", trial_name="b", config={}, _wandb=MagicMock())
+    assert recorded[TagKey.AIR_SETUP_WANDB_INTEGRATION_USED] == "1"
+
+
+def test_tag_setup_mlflow(monkeypatch):
+    from ray.air.integrations.mlflow import setup_mlflow
+
+    recorded = _mock_record_from_module(
+        ray.air.integrations.mlflow, monkeypatch=monkeypatch
+    )
+    monkeypatch.setattr(ray.air.integrations.mlflow, "_MLflowLoggerUtil", MagicMock())
+    setup_mlflow()
+    assert recorded[TagKey.AIR_SETUP_MLFLOW_INTEGRATION_USED] == "1"
+
+
+@pytest.mark.parametrize(
+    "callback_classes",
+    [
+        None,
+        [],
+        [lambda: None],
+        DEFAULT_CALLBACK_CLASSES,
+        [
+            wandb.WandbLoggerCallback,
+            mlflow.MLflowLoggerCallback,
+            comet.CometLoggerCallback,
+            AimLoggerCallback,
+        ]
+        + [_CustomLoggerCallback, _CustomCallback] * 2,
+    ],
+)
+def test_tag_callbacks(mock_record, callback_classes):
+
+    callbacks = (
+        [callback_cls() for callback_cls in callback_classes]
+        if callback_classes
+        else None
+    )
+
+    recorded = air_usage.tag_callbacks(callbacks)
+    expected_callback_counts = air_usage._count_callbacks(callbacks)
```

Comment:
Hm I think it might be preferable to just hardcode the expected results in the parametrized input since we also want to test that`_count_callbacks` works as expected.